### DISCUSSION TITLE
Need help in Pycaret

### DISCUSSION THREAD

**Discussion by Nisharani123** [pycaret/pycaret#4050] [General]
I'm trying to work with Pycaret. I have an IR data set that I want to classify using Pycaret (different classification models). I have an IR data Excel sheet with 1668 columns and 8 rows. How can I use this data? Please let me know if anyone can help. I have used chemometrics methods earlier. Just wanted to explore Pycaret so I can use this in my research work.

**Comment by celestinoxp** [comment]
I can try to help, but I need some details, is the problem classification or regression?
assuming it is classification you can see here: https://pycaret.readthedocs.io/en/latest/api/classification.html

to import excel file you can use something like

```
import pandas as pd

# Upload the Excel file
df = pd.read_excel('path/to/your/file.xlsx')
```

**Reply by Nisharani123** [reply]
I have imported my IR data from unscramblerX (a chemometrics software). Now I have an Excel sheet of 1668 columns and 8 rows with me. For classification I need to set target data in which I have to put names of the columns (that is what I have learned till now ) but my data only contains numerical values. I want to know how can I set target variables? How can I create training and validation dataset.

**Reply by celestinoxp** [reply]
The advantage of pycaret is that it does all the preprocessing automatically, and you don't need to divide training and testing, it does it internally, so you just need to define what the target is in setup(...)

if interested i can explain in slack https://join.slack.com/t/pycaret/shared_invite/zt-2p3r5unxu-tggVBMiZCCSjh2dM5UKUMQ
